mittab/apps/tab/outround_pairing_views.py

```diff
@@ -0,0 +1,639 @@
+
+    return render(request, "pairing/pair_round.html", locals())
+
+
+def outround_pairing_view(request,
+                          type_of_round=BreakingTeam.VARSITY,
+                          num_teams=None):
+
+    if num_teams is None:
+        num_teams = TabSettings.get("var_teams_to_break", 8)
+
+        while not math.log(num_teams, 2) % 1 == 0:
+            num_teams += 1
+
+        return redirect("outround_pairing_view",
+                        type_of_round=BreakingTeam.VARSITY,
+                        num_teams=num_teams)
+
+    pairing_released = False
+
+    if type_of_round == BreakingTeam.VARSITY:
+        pairing_released = TabSettings.get("var_teams_visible", 256) <= num_teams
+    elif type_of_round == BreakingTeam.NOVICE:
+        pairing_released = TabSettings.get("nov_teams_visible", 256) <= num_teams
+
+    label = "[%s] Ro%s" % ("V" if type_of_round == BreakingTeam.VARSITY else "N",
+                           num_teams)
+    outround_options = []
+
+    var_teams_to_break = TabSettings.get("var_teams_to_break", 8)
+    nov_teams_to_break = TabSettings.get("nov_teams_to_break", 4)
+
+    while not math.log(var_teams_to_break, 2) % 1 == 0:
+        var_teams_to_break += 1
+
+    while not math.log(nov_teams_to_break, 2) % 1 == 0:
+        nov_teams_to_break += 1
+
+    while var_teams_to_break > 1:
+        if Outround.objects.filter(type_of_round=BreakingTeam.VARSITY,
+                                   num_teams=var_teams_to_break).exists():
+            outround_options.append(
+                (reverse("outround_pairing_view", kwargs={
+                    "type_of_round": BreakingTeam.VARSITY,
+                    "num_teams": int(var_teams_to_break)}),
+                 "[V] Ro%s" % (int(var_teams_to_break),))
+            )
+        var_teams_to_break /= 2
+
+    while nov_teams_to_break > 1:
+        if Outround.objects.filter(type_of_round=BreakingTeam.NOVICE,
+                                   num_teams=nov_teams_to_break).exists():
+            outround_options.append(
+                (reverse("outround_pairing_view", kwargs={
+                    "type_of_round": BreakingTeam.NOVICE,
+                    "num_teams": int(nov_teams_to_break)}),
+                 "[N] Ro%s" % (int(nov_teams_to_break),))
+            )
+        nov_teams_to_break /= 2
+
+    outrounds = Outround.objects.filter(type_of_round=type_of_round,
+                                        num_teams=num_teams).all()
+
+    judges_per_panel = TabSettings.get("var_panel_size", 3) \
+                       if type_of_round == BreakingTeam.VARSITY \
+                          else TabSettings.get("nov_panel_size", 3)
+    judge_slots = [i for i in range(1, judges_per_panel + 1)]
+
+    var_to_nov = TabSettings.get("var_to_nov", 2)
+    other_round_num = num_teams / var_to_nov
+    if type_of_round == BreakingTeam.NOVICE:
+        other_round_num = num_teams * var_to_nov
+
+    other_round_type = not type_of_round
```

Comment:
IMO it's worth being a bit more verbose here for clarity:

```suggestion
    other_round_type = BreakingTeam.NOVICE if type_of_round == BreakingTeam.VARSITY else BreakingTeam.VARSITY
    ```
```

Reply:
I think I got all instances of this in recent commit